src/components/logic/log_skeleton.py

```diff
@@ -0,0 +1,68 @@
+"""Generates a log skeleton model based on the underlying relationships."""
+
+import os
+from src.components.logic import relationships as rel
+from src.components.util.xes_importer import XES_Importer
+from pm4py.algo.discovery.log_skeleton import algorithm as lsk_discovery
+
+
+class Log_Skeleton:
+    """Class that combines all relationships and generates a log skeleton."""
+
+    def __init__(self, log, all_activities, noise_threshhold):
+        """Init an instance that will generate a log skeleton model.
+
+        Parameters:
+            log : Log of traces
+            all_activities : Collection of all occuring activities
+            noise_threshhold : Set sentivity level for the algorithms
+        """
+        self.log = log
+        self.relationships = {
+            # Dictionary contains string representation of all classes
+            rel.Always_Before: 'always_before',
+            rel.AlwaysAfter: 'always_after',
+            rel.Equivalence: 'equivalence',
+            rel.Never_Together: 'never_together',
+            rel.Next_Both_Ways: 'next_both_ways',
+            rel.Next_One_Way: 'next_one_way',
+            rel.Counter: 'counter',
+        }
+        self.all_activities = all_activities
+        self.noise_threshhold = noise_threshhold
+
+    def apply(self):
+        """Return the log skeleton model as a dictionary.
+
+        Each key contains name of the relationship.
+        The corresponding value is the set with all
+        activitiy pairs of that relationship.
+        """
+        res = {}
+        for r in self.relationships:
+            r_instance = r(self.log, self.all_activities,
+                           self.noise_threshhold,
+                           rel.Relationship.Mode.FORALL, False)
```

Comment:
These parameters are not necessary they default to these values. Can you replace that by:

```           
 r_instance = r(self.log, self.all_activities,
                          self.noise_threshhold)
```

Oh and it's _noise_threshold_ not _noise_threshhold_ just a typo. 😅